miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/api/ApiClientSpec.kt

```diff
@@ -126,14 +127,28 @@ open class RetrofitRequestExecutorNormalSpec : RetrofitRequestExecutorSpec() {
 
 open class RetrofitRequestExecutorErrorSpec : RetrofitRequestExecutorSpec() {
 
-    @Test(expected = MiniAppHttpException::class)
+    @Test(expected = MiniAppSdkException::class)
     fun `should throw when server returns error response`() = runBlockingTest {
         mockServer.enqueue(createErrorResponse())
 
         createRequestExecutor()
             .executeRequest(createApi().fetch())
     }
 
+    @Test
+    fun `should throw exception when the response returned by server is not of type T`() =
+        runBlockingTest {
+            mockServer.enqueue(createInvalidTestApiResponse())
+            try {
+                createRequestExecutor()
+                    .executeRequest(createApi().fetch())
+
+                TestCase.fail("Should have thrown MiniAppSdkException.")
+            } catch (exception: MiniAppSdkException) {
```

Comment:
You should be able to use the `expected` parameter instead of try/catch for this test case: `@Test(expected = MiniAppSdkException::class)`